test/misc.jl

```diff
@@ -7,3 +7,13 @@ using Pkg
     f() = Pkg.Types.UNREGISTERED_STDLIBS
     @inferred f()
 end
+
+@testset "hashing" begin
+    @test hash(Pkg.Types.Project()) == 0xd4b8cda960e4de88
+    @test hash(Pkg.Types.VersionBound()) == 0x873f51b414a315c2
+    @test hash(Pkg.Resolve.Fixed(VersionNumber(0,1,0))) == 0xea3ee20b2c9d6ccb
+
+    # hashes for these empty objects aren't stable
+    hash(Pkg.Types.VersionSpec())
+    hash(Pkg.Types.PackageEntry())
```

Comment:
I don't know why, but these hash test results aren't stable

Reply:
Can you elaborate on what you mean by this?

Reply:
They don't give the same result each time

Reply:
@StefanKarpinski Any idea why this might be?

Reply:
> Huh. That seems.... bad?

Why, there is nothing saying that hashes need to be stable over julia sessions.

Reply:
They are `mutable struct`s:
```
julia> struct X x end

julia> mutable struct Y y end

julia> hash(X(1))
0x2211c0ec74129abf

julia> hash(X(1))
0x2211c0ec74129abf

julia> hash(Y(1))
0xeb1f48179582169f

julia> hash(Y(1))
0x006bb8c4d822c6fb
```